packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -290,6 +296,8 @@ export default compose( [
 		const isInnerBlockHolder = name === getGroupingBlockName();
 		const isRootListInnerBlockHolder = ! isSelectedBlockNested && isInnerBlockHolder;
 
+		const shouldApplyVerticalMarginStyle = ! isLastBlock && ( ( isFirstBlock && parentCount === 2 ) || parentCount > 2 );
```

Comment:
If I understand this correctly, this is trying to add a margin to every sibling except the last one, right?

This expression seems more complicated than it needs, if I understand each variable correctly. I see three possible cases regarding the number of children in the parent: 

- There's one block (`parentCount < 2`). Then this must be the last (and first) block, so `! isLastBlock === false` and the margin shouldn't apply.
- There are two blocks (`parentCount == 2`). We have already set a condition to not add a margin on the last block, so the `isFirstBlock` is only relevant for the other case, which will always be true: if I'm not the last of 2, I must be the first.
- There are more than two blocks (`parentCount > 2`). Similarly, we've ruled out the case where we are the last block, and we want to add the margin for every other block.

So, if I'm not missing anything, the same logic should work with this simpler condition:

```suggestion
		const shouldApplyVerticalMarginStyle = ! isLastBlock;
```

Reply:
I have check it and it's fine but after we modify `isLastBlock` as below:

<image src="https://user-images.githubusercontent.com/21242757/73541574-1ab03d80-4433-11ea-8d86-0e1e52a097c7.png" width="500" />

In previous implementation it tries to check the order base of current `clientId` InnerBlocks count. Which isn't proper because we need to check the order in parent InnerBlock list.

I have checked and it seems that this flag isn't use anywhere so probably we can also prevent to pass it to component props

Reply:
I have also simplify passing the prop in latest commit